src/group/mod.rs

```diff
@@ -0,0 +1,188 @@
+// Copyright 2019 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! The module includes the definition of new feature 'raft group' which is used for
+//! **Follower Replication**
+//!
+//! # Follower Replication
+//! See https://github.com/tikv/rfcs/pull/33
+
+use std::collections::hash_map::Iter;
+use std::collections::HashMap;
+use std::iter::FromIterator;
+
+/// The option used for feature [Follower Replication](https://github.com/tikv/rfcs/pull/33/files)
+#[derive(Clone, Debug)]
+pub struct FollowerReplicationOption {
+    /// The leader choose a follower in a raft group as a delegate and ask it to send entries to
+    /// the rest group members.
+    /// If there is no group configured, no delegate will be picked
+    pub follower_delegate: bool,
+
+    /// The raft group definition. See [`GroupsConfig`](group/struct.GroupsConfig.html) for detail.
+    pub groups: GroupsConfig,
+
+    /// The max number of members of a group of which contains leader and the leader never pick a delegate.
+    /// If the group size is larger than this, a delegate will be picked even the leader belongs to this group.
+    pub max_leader_group_no_delegate: usize,
```

Comment:
Personally I prefer to remove this, because without explicit information about `group_id`, follower replication may be not helpful. For example 7 peers in one datacenter, follower replication can't save any network traffics.